node/src/components/linear_chain_sync.rs

```diff
@@ -315,23 +424,19 @@ where
                         // `block_height` not found on any of the peers.
                         // We have synchronized all, currently existing, descendants of trusted
                         // hash.
-                        effect_builder
-                            .immediately()
-                            .event(move |_| Event::StartDownloadingDeploys)
+                        self.mark_done();
+                        Effects::new()
                     }
                     Some(peer) => fetch_block_at_height(effect_builder, peer, block_height),
                 },
                 BlockByHeightResult::FromStorage(block) => {
-                    self.state.block_downloaded(&block);
-                    // We should be checking the local storage for linear blocks before we
-                    // start syncing.
+                    // We shouldn't get invalid data from the storage.
+                    // If we do, it's a bug.
```

Comment:
(…or the user messed with the storage or something. In the very long term we might not want to panic here; but for now I agree.)

Reply:
if a user messed up with the storage then it's even worse.